x/custom/model.go

```diff
@@ -0,0 +1,102 @@
+package custom
+
+import (
+	"github.com/iov-one/tutorial/morm"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+)
+
+var _ morm.Model = (*CustomStateIndexed)(nil)
+
+func (m *CustomStateIndexed) SetID(id []byte) error {
+	m.ID = id
+	return nil
+}
+
+func (m *CustomStateIndexed) Validate() error {
+	var errs error
+
+	errs = errors.AppendField(errs, "Metadata", m.Metadata.Validate())
+	errs = errors.AppendField(errs, "ID", isGenID(m.ID, true))
+	errs = errors.AppendField(errs, "CustomString", customStringValidation(m.CustomString))
+	if m.CustomByte == nil {
+		errs = errors.Append(errs, errors.Field("CustomByte", errors.ErrEmpty, "required"))
+	}
+	if m.InnerStateEnum != InnerStateEnum_CaseOne && m.InnerStateEnum != InnerStateEnum_CaseTwo {
+		errs = errors.Append(errs,
+			errors.Field("InnerStateEnum", errors.ErrState, "invalid"))
```

Comment:
I think now that you do not include any specific information in the description, you can skip it. Use `errors.AppendField` instead of `errors.Append` and `errors.Field`. 

[`ErrState` descirption](https://github.com/iov-one/weave/blob/master/errors/errors.go#L47) is _invalid state_, so adding _invalid_ does not provide any information.

Reply:
:+1: I should go through all error descriptions

Reply:
And I think this is same for most of the `errors.ErrEmpty` validation errors.